tuta-sdk/rust/sdk/src/entities/entity_facade.rs

```diff
@@ -41,15 +62,238 @@ pub trait EntityFacade: Send + Sync {
+			{
+				let final_iv = Iv::from_bytes(
+					Self::get_final_iv_for_key(instance, key)
+						.unwrap()
+						.assert_bytes()
+						.as_slice(),
+				)
+				.map_err(|err| ApiCallError::internal(format!("iv of illegal size {:?}", err)))?;
+				encrypted_value = Self::encrypt_value(
+					key,
+					model_value,
+					instance_value,
+					&sk,
+					IvProvider::Fixed(final_iv),
+				)?
+			} else {
+				encrypted_value = Self::encrypt_value(
+					key,
+					model_value,
+					instance_value,
+					&sk,
+					random_iv_provider.clone(),
+				)?
+			}
+			encrypted.insert(key.to_string(), encrypted_value);
+		}
+
+		if type_model.element_type == ElementType::Aggregated && !encrypted.contains_key("_id") {
+			let randomizer = RandomizerFacade::from_core(rand_core::OsRng);
+			let new_id = randomizer.generate_random_array::<4>();
+
+			encrypted.insert(
+				String::from("_id"),
+				ElementValue::String(BASE64_URL_SAFE_NO_PAD.encode(BASE64_STANDARD.encode(new_id))),
+			);
+		}
+
+		for (association_name, association) in &type_model.associations {
+			if let AssociationType::Aggregation = association.association_type {
+				let dependency = association.dependency.unwrap_or(type_model.app);
+				let aggregated_type_model = self
+					.type_model_provider
+					.get_type_model(dependency, association.ref_type)
+					.unwrap_or_else(|| {
+						panic!(
+							"Cannot find type model for: {:?}",
+							(dependency, association.ref_type)
+						)
+					});
+				let aggregation = association;
+				let instance_association = instance.get(&association_name.to_string()).unwrap();
+				if aggregation.cardinality == Cardinality::ZeroOrOne
+					&& instance_association == &ElementValue::Null
+				{
+					encrypted.insert(association_name.to_string(), ElementValue::Null);
+				} else if instance_association == &ElementValue::Null {
+					panic!("Undefined attribute {}:{association_name}", type_model.name);
+				} else if aggregation.cardinality == Cardinality::Any {
+					let aggregates = instance_association.assert_array();
+					let mut encrypted_aggregates = Vec::with_capacity(aggregates.len());
+					for aggregate in &aggregates {
+						let parsed_entity = self.encrypt_and_map_to_literal_with_iv(
+							aggregated_type_model,
+							&aggregate.assert_dict(),
+							sk.clone(),
+							random_iv_provider.clone(),
+						)?;
+						encrypted_aggregates.push(ElementValue::Dict(parsed_entity));
+					}
+
+					encrypted.insert(
+						association_name.to_string(),
+						ElementValue::Array(encrypted_aggregates),
+					);
+				} else {
+					let parsed_entity = self.encrypt_and_map_to_literal_with_iv(
+						aggregated_type_model,
+						&instance_association.assert_dict(),
+						sk.clone(),
+						random_iv_provider.clone(),
+					)?;
+					let encrypted_aggregate = ElementValue::Dict(parsed_entity);
+					encrypted.insert(association_name.to_string(), encrypted_aggregate);
+				}
+			} else {
```

Comment:
I think this `if-else` is too big

Reply:
Splitted the association == Aggregated to seperate function now